kratos/includes/model_part.h

```diff
@@ -568,6 +599,111 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     }
 
 
+    ///@}
+    ///@name MasterSlaveConstraints
+    ///@{
+
+    SizeType NumberOfMasterSlaveConstraints() const
+    {
+        return mMasterSlaveConstraints.size();
+    }
+
+    MasterSlaveConstraintContainerType& MasterSlaveConstraints()
+    {
+        return mMasterSlaveConstraints;
+    }
+
+    MasterSlaveConstraintContainerType::Pointer pGetMasterSlaveConstraints()
+    {
+        return Kratos::shared_ptr<MasterSlaveConstraintContainerType>(&mMasterSlaveConstraints);
```

Comment:
I don't think you should create a shared pointer like this.

Reply:
This functions is necessary basically for making the MasterSlaveConstraints a property of ModelPart when adding it to python. I will see if I can replace it. 
Is there any other way to so this pointer creation.

Reply:
The problem is shared_ptr is for heap allocated objects. And even if mMasterSlaveConstraints were heap allocated, one should not create a shared_ptr from raw ptr like this as the count will be wrong. As soon as that ptr goes out of scope something bad will happen.

Reply:
about the python interface `mMasterSlaveConstraints` exists and has a owner who manages its lifetime no?

if that is so, why don't you give back to python a reference instead of a Pointer? (return_internal_reference)